.github/workflows/dev-workflow.yaml

```diff
@@ -0,0 +1,241 @@
+
+  determine_pr_size:
+    name: Determine the size of the PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: codelytv/pr-size-labeler@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          xs_label: 'size: XS'
+          xs_max_size: '30'
+          s_label: 'size: S'
+          s_max_size: '100'
+          m_label: 'size: M'
+          m_max_size: '500'
+          l_label: 'size: L'
+          l_max_size: '1000'
+          xl_label: 'size: XL'
+          fail_if_xl: 'false'
+          message_if_xl: >
+            'This PR exceeds the recommended size of 1000 lines.
+            Please make sure you are NOT addressing multiple issues with one PR.
+            Note this PR might be rejected due to its size.’
+  bump-version:
+    name: Bump app version using PR labels
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Bump version
+        uses: apomalyn/bump-version-using-labels@1.3.1
+        with:
+          file_path: 'pubspec.yaml'
+          reference_branch: 'master'
+  ##############################################################
+  # Testing
+  ##############################################################
+  testing:
+    name: Tests and checks
+    runs-on: ubuntu-latest
+    outputs:
+      coverage: ${{ steps.coverage.outputs.percentage }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: '3.3.x'
+          channel: 'stable'
+          cache: true
+      - run: flutter doctor
+      - name: Decrypt SignETS certificate and Google Services files
+        run: |
+          chmod +x ./scripts/decrypt.sh
+          ./scripts/decrypt.sh
+        env:
+          ENCRYPTED_SIGNETS_API_CERT_PASSWORD: ${{ secrets.ENCRYPTED_SIGNETS_API_CERT_PASSWORD }}
+          ENCRYPTED_GOOGLE_SERVICE_PASSWORD: ${{ secrets.ENCRYPTED_GOOGLE_SERVICE_PASSWORD }}
+
+      # Get flutter dependencies.
+      - run: flutter pub get
+
+      # Check the format of the code and commit the formatted files.
+      - name: Format files in lib and test directories
+        run: flutter format lib test
+      - name: Commit formatted files
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          file_pattern: "*.dart"
+          commit_user_name: github-actions[bot]
+          commit_user_email: 41898282+github-actions[bot]@users.noreply.github.com
+          commit_message: "[BOT] Applying format."
+          add_options: '-u'
+
+      # Check if the code has any errors/warnings
+      - name: Analyze code
+        run: flutter analyze
+
+      # Run the tests with --update-goldens.
+      - name: Run tests
+        run: flutter test --coverage --update-goldens
```

Comment:
Are you sure you want to update the goldens at every PR? If yes, I think you can simply remove the goldens tests

Reply:
The objective of updating the goldens at every PR is to be able to see the UI changes for all PR. By doing that, we can see if some changes to the UI were unintentional. We could also add a label to update goldens if you think it’s necessary.

Reply:
Instead of doing a commit, you could comment on the PR if a change is detected and out the goldens as an artifact on the workflow.

Let me give an example of a problem I see, you create a PR, the goldens are modified and multiples reviewers asks for some changes (not related to the goldens). A discussion starts, you applies the changes and the PR receive something like 10 new commits. By now there is a good chances that everyone forgot about the goldens that where modified (especially if this is a capricious one who change regularly for nothing like QuickLinks in the past)

If everyone is okay with the proposed behavior (I'm speaking about the PR proposed changes not your discussion) it's fine by me ^^ I just wanted to point that ^^

Reply:
Moi je suis d'accord! Je trouve que c'est un bon conpromis entre tout retirer les goldens et laisser l'état actuel de gestion des goldens... Il y a beaucoup de problème au niveau du développement actuellement concernant les goldens notamment pour les nouveaux membres. Je trouve que ça décourage beaucoup au développement de nouvelles fonctionnalités. Mais je comprends le problème, il va juste falloir que les gens soit plus à l'affût avant de mettre un approved! Je vais leur en faire part!